addons/wiki/migrations/0007_auto_20180124_1152.py

```diff
@@ -0,0 +1,191 @@
+    desired_wiki_id = get_guid(state, desired_wiki_object)._id
+    for contrib in Contributor.objects.filter(node=node):
+        if contrib.user.comments_viewed_timestamp.get(current_wiki_id, None):
+            timestamp = contrib.user.comments_viewed_timestamp[current_wiki_id]
+            contrib.user.comments_viewed_timestamp[desired_wiki_id] = timestamp
+            del contrib.user.comments_viewed_timestamp[desired_wiki_id]
+            contributors_pending_save.append(contrib)
+    bulk_update(contributors_pending_save)
+    return
+
+def migrate_guid_referent(state, guid_id, desired_referent):
+    """
+    Point the guid towards the desired_referent.
+    Pointing the NodeWikiPage guid towards the WikiPage will still allow links to work.
+    """
+    Guid = state.get_model('osf', 'guid')
+    guid = Guid.objects.get(_id=guid_id)
+    guid.content_type_id = ContentType.objects.get_for_model(desired_referent).id
+    guid.object_id = desired_referent.id
+    return guid
+
+def create_wiki_page(state, node, node_wiki, page_name):
+    WikiPage = state.get_model('addons_wiki', 'wikipage')
+    return WikiPage(
+        page_name=page_name,
+        user=node_wiki.user,
+        node=node,
+        created=node_wiki.created,
+        modified=node_wiki.modified,
+    )
+
+def create_wiki_version(state, node_wiki, wiki_page):
+    WikiVersion = state.get_model('addons_wiki', 'wikiversion')
+    return WikiVersion(
+        wiki_page=wiki_page,
+        user=node_wiki.user,
+        created=node_wiki.created,
+        modified=node_wiki.modified,
+        content=node_wiki.content,
+        identifier=node_wiki.version,
+    )
+
+def create_guids(state):
+    WikiPage = state.get_model('addons_wiki', 'wikipage')
+    Guid = state.get_model('osf', 'guid')
+    guid_count = 0
+    content_type = ContentType.objects.get_for_model(WikiPage)
+    wp_count = WikiPage.objects.count()
+    for wp in WikiPage.objects.all():
+        # looping instead of bulk_create, so _id's are not the same
+        guid_count += 1
+        print 'Guid: {}/{}'.format(guid_count, wp_count)
+        Guid.objects.create(object_id=wp.id, content_type_id=content_type.id)
+    return
+
+def create_wiki_pages(state, nodes):
+    Guid = state.get_model('osf', 'guid')
+    WikiPage = state.get_model('addons_wiki', 'wikipage')
+    NodeWikiPage = state.get_model('addons_wiki', 'nodewikipage')
+    WikiPage = state.get_model('addons_wiki', 'wikipage')
+    node_total = nodes.count()
+    node_count = 0
+    wiki_pages = []
+    print 'Creating wiki pages'
+    for node in nodes:
+        node_count += 1
+        print 'NODE: {}/{}'.format(node_count, node_total)
+        for wiki_key, version_list in node.wiki_pages_versions.iteritems():
+            node_wiki = NodeWikiPage.objects.get(id=Guid.objects.get(_id=version_list[0]).object_id)
+            latest_page_name = NodeWikiPage.objects.get(id=Guid.objects.get(_id=version_list[-1]).object_id).page_name
```

Comment:
I originally did it like this `guids____id=...` but because I switched to loading models with `state.get_model` this doesn't work anymore.  Matt said it was because these are created dynamically and the guid part isn't included with it.  Our discussion here https://github.com/CenterForOpenScience/osf.io/pull/8117#discussion_r165487643

Reply:
Ah, I figured `.load()` wouldn't work but I didn't think `guids___id=` would also be broken. I guess that makes sense since the `guids` GenericRelation is mixed in. OK, then querying like this is fine. Though you can still use `.values_list` for a minor speedup.

Reply:
yes, thanks for the tip!